16beb84bca4953092b9ff3f159575f08250c2790
src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java
--- 
+++ 
@@ -36,6 +36,7 @@
 import com.yelp.nrtsearch.server.luceneserver.DeleteDocumentsHandler;
 import com.yelp.nrtsearch.server.luceneserver.DeleteIndexHandler;
 import com.yelp.nrtsearch.server.luceneserver.GetNodesInfoHandler;
+import com.yelp.nrtsearch.server.luceneserver.GetStateHandler;
 import com.yelp.nrtsearch.server.luceneserver.GlobalState;
 import com.yelp.nrtsearch.server.luceneserver.IndexState;
 import com.yelp.nrtsearch.server.luceneserver.LiveSettingsHandler;
@@ -795,6 +796,24 @@
             }
         }
 
+        @Override
+        public void state(StateRequest request, StreamObserver<StateResponse> responseObserver) {
+            try {
+                IndexState indexState = globalState.getIndex(request.getIndexName());
+                StateResponse reply = new GetStateHandler().handle(indexState, request);
+                logger.info("IndicesRequestHandler returned " + reply.toString());
+                responseObserver.onNext(reply);
+                responseObserver.onCompleted();
+            } catch (Exception e) {
+                logger.warn("error while trying to get state for index " + request.getIndexName(), e);
+                responseObserver.onError(Status
+                        .INVALID_ARGUMENT
+                        .withDescription("error while trying to get state for index " + request.getIndexName())
+                        .augmentDescription(e.getMessage())
+                        .asRuntimeException());
+            }
+        }
+
     }
 
     static class ReplicationServerImpl extends ReplicationServerGrpc.ReplicationServerImplBase {